kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -351,19 +346,35 @@ public void initiateConnect(HostPort remote, List<FilterAndInvoker> filters) {
      * Called when the channel to the upstream broker is failed.
      */
     @VisibleForTesting
-    void onUpstreamChannelFailed(HostPort remote, Future<?> future, Channel inboundChannel) {
+    void onUpstreamChannelFailed(HostPort remote, Throwable upstreamFailure) {
         state = State.FAILED;
         // Close the connection if the connection attempt has failed.
-        Throwable failureCause = future.cause();
-        exceptionHandler.mapException(failureCause).ifPresentOrElse(
-                result -> closeWith(inboundChannel, result),
-                () -> {
-                    LOGGER.atWarn()
-                            .setCause(LOGGER.isDebugEnabled() ? failureCause : null)
-                            .log("Connection to target cluster on {} failed with: {}, closing inbound channel. Increase log level to DEBUG for stacktrace",
-                                    remote, failureCause.getMessage());
-                    closeNoResponse(inboundChannel);
-                });
+        Throwable errorCodeEx = null;
+        if (upstreamFailure instanceof SSLHandshakeException e) {
```

Comment:
Its not this simple as this at least in some situations that we care about the SSLHandshakeException is wrapped in a DecoderException.